src/main/java/org/opentripplanner/middleware/trip_monitor/jobs/CheckMonitoredTrip.java

```diff
@@ -0,0 +1,335 @@
+
+import static org.opentripplanner.middleware.trip_monitor.jobs.NotificationType.ARRIVAL_DELAY;
+import static org.opentripplanner.middleware.trip_monitor.jobs.NotificationType.DEPARTURE_DELAY;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getZoneIdForCoordinates;
+
+/**
+ * This job handles the primary functions for checking a {@link MonitoredTrip}, including:
+ * - determining if a check should be run (based on mostly date/time),
+ * - making requests to OTP and comparing the stored itinerary against these new responses from OTP, and
+ * - determining if notifications should be sent to the user monitoring the trip based on their saved criteria.
+ */
+public class CheckMonitoredTrip implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(CheckMonitoredTrip.class);
+    private final MonitoredTrip trip;
+    public int departureDelay;
+    public int arrivalDelay;
+    /**
+     * This is only used during testing to inject a mock OTP response for comparison against a monitored trip.
+     */
+    private OtpDispatcherResponse injectedOtpResponseForTesting;
+    /**
+     * Used to track the various check trip notifications and construct email/SMS messages.
+     */
+    public final Set<TripMonitorNotification> notifications = new HashSet<>();
+    /**
+     * Whether the check was skipped based on time/date criteria.
+     */
+    public boolean checkSkipped;
+    /**
+     * The index of the matching {@link Itinerary} as found in the OTP {@link Response} planned as part of this check.
+     */
+    public int matchingItineraryIndex = -1;
+    /**
+     * The OTP response planned to check the stored {@link Itinerary} against.
+     */
+    public Response otpResponse;
+    /**
+     * Tracks the time the notification was sent to the user.
+     */
+    public long notificationTimestamp = -1;
+
+    public CheckMonitoredTrip(MonitoredTrip trip) {
+        this.trip = trip;
+    }
+
+    /**
+     * This constructor should only be used for testing when we need to inject a mock OTP response.
+     */
+    public CheckMonitoredTrip(MonitoredTrip trip, OtpDispatcherResponse injectedOtpResponseForTesting) {
+        this.trip = trip;
+        this.injectedOtpResponseForTesting = injectedOtpResponseForTesting;
+    }
+
+    @Override
+    public void run() {
+        // Check if the trip check should be skipped (based on time, day of week, etc.)
+        checkSkipped = shouldSkipMonitoredTripCheck(trip);
+        if (checkSkipped) {
+            LOG.debug("Skipping check for trip: {}", trip.id);
+            return;
+        }
+        // Make a request to OTP with the monitored trip params or if this is in a test environment use the injected OTP
+        // response.
+        OtpDispatcherResponse otpDispatcherResponse = injectedOtpResponseForTesting != null
+            ? injectedOtpResponseForTesting
+            : OtpDispatcher.sendOtpPlanRequest(trip.queryParams);
```

Comment:
This will keep requesting the same trip plan on the same date. I am working on a fix that will inject the proper date.

Reply:
I pushed some changes that introduce a framework for updating the date parameter, but didn't have time to code and test the calculations of the proper generation of the date. Furthermore, that implementation depends on what we decide we want to do regarding https://github.com/ibi-group/otp-middleware/pull/33#discussion_r469026603.